lib/checker.js

```diff
@@ -1673,6 +1666,28 @@ function typesMatch(type1,type2) {
 	);
 }
 
+function isAssignmentAllowed(sourceType, targetType){
+	if (typesMatch(sourceType, targetType)){
+		return true;
+	}
+
+	let matches = {
+		int: ["int"],
+		number: ["number", "int", "finite"],
+		finite: ["finite", "int"],
+		bint: ["bint", "int"]
+	}
+	
+	let sourceTypeID = getTypeID(sourceType) 
+	let targetTypeID = getTypeID(targetType) 
+
+	return (
+		matches[targetTypeID] && 
+		matches[targetTypeID].includes(sourceTypeID) ||
+		targetTypeID == "any"
```

Comment:
The && and || chained together seems to create a bit of confusion here. I think we could take out the "any" check here and just have an `if` statement up on line 1673, and that would be clearer about that special case.